src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -254,21 +254,29 @@ public function getManipulatedData(GridField $gridField, SS_List $dataList)
     public function canFilterAnyColumns($gridField)
     {
         $list = $gridField->getList();
-
-        if (!$this->checkDataType($list)) {
+        if (!($list instanceof Filterable) || !$this->checkDataType($list)) {
             return false;
         }
-
-        $columns = $gridField->getColumns();
-        foreach ($columns as $columnField) {
-            $metadata = $gridField->getColumnMetadata($columnField);
-            $title = $metadata['title'];
-
-            if ($title && $list->canFilterBy($columnField)) {
-                return true;
+        $modelClass = $gridField->getModelClass();
+        $searchableFields = Config::inst()->get($modelClass, 'searchable_fields');
+        if (!empty($searchableFields)) {
```

Comment:
Do we want to make sure here that `searchable_fields` is an array if set?

Reply:
Have changed to use $modelClass::singleton()->searchableFields() which returns an array